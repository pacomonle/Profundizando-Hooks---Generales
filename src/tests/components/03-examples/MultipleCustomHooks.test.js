import React from 'react';
import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import  MultipleCustomHooks  from '../../../components/03-examples/MultipleCustomsHooks';

import  useFetch  from '../../../hooks/useFetch';
import  useCounter  from '../../../hooks/useCounter';

// implementar mock de hooks
jest.mock('../../../hooks/useFetch');
jest.mock('../../../hooks/useCounter');




describe('Pruebas en <MultipleCustomHooks />', () => {
// mockear lo que devuelve un hook
    useCounter.mockReturnValue({
        state: 10,
        increment: () => {},
        decrement: () => {}
    });
    
    
    test('debe de mostrarse correctamente', () => {
  // mockear lo que devuelve un hook
        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks></MultipleCustomHooks>);
        expect( wrapper ).toMatchSnapshot();
        
    })

    test('should debe de mostrar la informaciÃ³n', () => {
       // mockear lo que devuelve un hook 
        useFetch.mockReturnValue({
            data: [{
                author: 'Fernando',
                quote: 'Hola Mundo'
            }],
            loading: false,
            error: null
        });

        const wrapper = shallow( <MultipleCustomHooks></MultipleCustomHooks>);
        // console.log(wrapper.html())
        expect( wrapper.find('p').at(1).exists() ).toBe( false );
        expect( wrapper.find('.mb-1').text().trim() ).toBe( 'Hola Mundo' );
        expect( wrapper.find('footer').text().trim() ).toBe( 'Fernando' );

    })
    
    

})


