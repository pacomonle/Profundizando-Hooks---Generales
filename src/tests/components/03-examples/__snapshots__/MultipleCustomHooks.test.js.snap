// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <MultipleCustomHooks /> debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Multiple Custom Hooks
  </h1>
  <h3>
    Breaking Bad Quotes
  </h3>
  <hr />
  <div
    className="alert alert-info"
  >
    <p>
      Loading....
    </p>
  </div>
  <button
    className="btn btn-primary btn-block"
    onClick={[Function]}
    type="button"
  >
    Siguiente frase
  </button>
  <button
    className="btn btn-warning btn-block mt-3"
    onClick={[Function]}
    type="button"
  >
    Anterior frase
  </button>
</div>
`;
